R/mle-finder.R

```diff
@@ -21,3 +21,76 @@ linear_BFGS <- function(design, outcome, noise_var = 1) {
   return(MLE)
 }
 
+logit_loglik <- function(coef, design, outcome) {
+
+  n <- length(outcome)
+  log_lik <- sum(sapply(1:n, function(i){
+    outcome[i] * sum(design[i,] * coef) -
+      log(1 + exp(sum(design[i,] * coef)))
+    }))
+  return(log_lik)
+
+}
+
+logit_gradient <- function(coef, design, outcome) {
+
+  pi <- exp(design %*% coef) / (1 + exp(design %*% coef))
```

Comment:
-1

Try not to overwrite existing variables like `pi`, which is defined as 3.14... already. Could use something like `prob`